cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -208,6 +209,11 @@ public void eventReceived(Action action, KafkaConnector kafkaConnector) {
                                                 log.info("{} {} in namespace {} was {}, but Connect cluster {} does not exist", connectorKind, connectorName, connectorNamespace, action, connectName);
                                                 updateStatus(noConnectCluster(connectNamespace, connectName), kafkaConnector, connectOperator.connectorOperator);
                                                 return Future.succeededFuture();
+                                            } else if ((connect != null && connect.getSpec() != null && connect.getSpec().getReplicas() == 0)
+                                                    || (connectS2i != null && connectS2i.getSpec() != null && connectS2i.getSpec().getReplicas() == 0))    {
+                                                log.info("{} {} in namespace {} was {}, but Connect cluster {} has 0 replicas", connectorKind, connectorName, connectorNamespace, action, connectName);
+                                                updateStatus(zeroReplicas(connectNamespace, connectName), kafkaConnector, connectOperator.connectorOperator);
+                                                return Future.succeededFuture();
                                             } else if (connect != null && isOlderOrAlone(connect.getMetadata().getCreationTimestamp(), connectS2i)) {
```

Comment:
What about the case where both connect and connectS2i are non null, but connectS2i is scaled to 0? We'd say there were zero replicas via the first branch, but we ought to proceed via the 2nd branch, right?

Reply:
Good point. I pushed it one block below to make sure it is reflected which Connect takes the priority. Thanks